plugins/hikvision/src/hikvision-camera-api.ts

```diff
@@ -32,7 +35,42 @@ export class HikVisionCameraAPI {
         });
     }
 
+    async checkDeviceModel() : Promise<string> {
+        if (!this.deviceModel) {
+            this.deviceModel = new Promise(async (resolve, reject) => {
+                try {
+                    const response = await this.digestAuth.request({
+                        method: "GET",
+                        responseType: 'text',
+                        url: `http://${this.ip}/ISAPI/System/deviceInfo`,
+                    });
+                    const deviceModel = response.data.match(/>(.*?)<\/model>/)?.[1];
+                    resolve(deviceModel);    
```

Comment:
I think another additional safe change could be made here: store this result in `this.storage.setItem('model', deviceModel)` and load it later. Can bypass this request forever once it succeeds. The model will also never change, obviously.

Reply:
I see.  Is this.storage uniqued to ip+user (since a user could have multiple DVRs)?

Reply:
this.storage is unique per device, yes.